src/main/java/com/ioet/bpm/skills/boundaries/CategoryController.java

```diff
@@ -0,0 +1,91 @@
+package com.ioet.bpm.skills.boundaries;
+
+import com.ioet.bpm.skills.domain.Category;
+import com.ioet.bpm.skills.repositories.CategoryRepository;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+
+    private final CategoryRepository categoryRepository;
+
+    public CategoryController(CategoryRepository categoryRepository) {
+        this.categoryRepository = categoryRepository;
+    }
+
+    @ApiOperation(value = "Return a list of all categorys", response = Category.class, responseContainer = "List")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Categorys successfully returned")
+    })
+    @GetMapping(produces = "application/json")
+    public ResponseEntity<Iterable> getAllCategories() {
+        Iterable<Category> categories = this.categoryRepository.findAll();
+        return new ResponseEntity<>(categories, HttpStatus.OK);
+    }
+
+    @ApiOperation(value = "Return one category", response = Category.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Category successfully returned")
+    })
+    @GetMapping(path = "/{id}", produces = "application/json")
+    public ResponseEntity<Category> getCategory(@PathVariable(value = "id") String categoryId) {
+        Optional<Category> categoryOptional = categoryRepository.findById(categoryId);
+        return categoryOptional.map(
+                category -> new ResponseEntity<>(category, HttpStatus.OK))
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+    }
+
+    @ApiOperation(value = "Create a new category", response = Category.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Category successfully created")
+    })
+    @PostMapping(produces = "application/json")    public ResponseEntity<Category> createCategory(@RequestBody Category category) {
```

Comment:
can we add a break line after the @PostMapping annotation?

Reply:
Done